integration/spark/src/main/java/io/openlineage/spark/agent/lifecycle/LogicalPlanSerializer.java

```diff
@@ -45,7 +45,9 @@ public static LogicalPlanSerializer getInstance() {
   private LogicalPlanSerializer() {
     mapper = new ObjectMapper();
     try {
-      mapper.registerModule(DefaultScalaModule$.MODULE$);
+      mapper.registerModule(
+          (Module)
+              Class.forName("com.fasterxml.jackson.module.scala.DefaultScalaModule").newInstance());
```

Comment:
These class names are different. Are we sure this works in both Spark versions?

Reply:
It's not working for spark3, due to `relocate` jackson libs.
Even if I get a `DefaultScalaModule$` from 2.12 , then If I try to register it  would receive an `ClassCastExeption`, that `com.fasterxml...Module `can.'t be cast to `openlineage.com.fasterxml...Module` That's why we need to have [PR](https://github.com/OpenLineage/OpenLineage/pull/275)